### ISSUE TITLE
"Trialwise Decoding" notebook does not give "stable" results

### ISSUE THREAD

**Issue by Venryx** [braindecode/braindecode#128]
(I'm new to machine-learning, so forgive me if this bug-report is misguided.)

I downloaded the "plot_bcic_iv_2a_moabb_trial.ipynb" example from [here](https://braindecode.org/auto_examples/plot_bcic_iv_2a_moabb_trial.html).

I noticed this code:
```
seed = 20200220  # random seed to make results reproducible
# Set random seed to be able to reproduce results
set_random_seeds(seed=seed, cuda=cuda)
```

Shouldn't this code make it so that every time I run the notebook, the results are the same in the learning-progress logging, and in the displayed plot?

However, it gives different results each time it runs, despite no code modifications: [Screen Capture](https://i.imgur.com/Q8MacPT.gif)

Questions:
1) Is it supposed to get different results each time?
2) If it is supposed to, is there any way I can modify the code to instead give "stable" results?

**Comment by Venryx** [comment]
Nevermind, I think I figured it out.

I was only rerunning the "training" block of code, which was not enough; I also had to re-run the "model creation" block, so that the model would get reset for each training run. (since the training block mutates the model)

To prevent me making this mistake again, I've now merged the model-creation and training code into the same code-block.

**Comment by robintibor** [comment]
Great, yes indeed, if you rerun the training code, you will keep training the already trained model while reinitializing the optimizer including learning rate etc. (which is a fine thing to do, similar to https://arxiv.org/abs/1608.03983) . glad you figure it out, let us know any further issues :)
